xds/src/main/java/io/grpc/xds/Bootstrapper.java

```diff
@@ -101,19 +101,19 @@ public static BootstrapInfo parseConfig(String rawData) throws IOException {
       logger.log(XdsLogLevel.INFO, "xDS server URI: {0}", serverUri);
       List<ChannelCreds> channelCredsOptions = new ArrayList<>();
       List<?> rawChannelCredsList = JsonUtil.getList(serverConfig, "channel_creds");
-      // List of channel creds is optional.
-      if (rawChannelCredsList != null) {
-        List<Map<String, ?>> channelCredsList = JsonUtil.checkObjectList(rawChannelCredsList);
-        for (Map<String, ?> channelCreds : channelCredsList) {
-          String type = JsonUtil.getString(channelCreds, "type");
-          if (type == null) {
-            throw new IOException("Invalid bootstrap: 'xds_servers' contains server with "
-                + "unknown type 'channel_creds'.");
-          }
-          logger.log(XdsLogLevel.INFO, "Channel credentials option: {0}", type);
-          ChannelCreds creds = new ChannelCreds(type, JsonUtil.getObject(channelCreds, "config"));
-          channelCredsOptions.add(creds);
+      if (rawChannelCredsList == null) {
+        throw new IOException("Invalid bootstrap: 'channel_creds' required");
+      }
+      List<Map<String, ?>> channelCredsList = JsonUtil.checkObjectList(rawChannelCredsList);
+      for (Map<String, ?> channelCreds : channelCredsList) {
+        String type = JsonUtil.getString(channelCreds, "type");
+        if (type == null) {
+          throw new IOException("Invalid bootstrap: 'xds_servers' contains server with "
+              + "unknown type 'channel_creds'.");
```

Comment:
This is the case of `type` not being set as opposed to "unknown type".

Reply:
What I mean here is for "type unspecified". Reading bootstrap file should not contain the logic for interpreting the type that gRPC supports, that should be done at the time when the channel creds is actually used.

Say the bootstrap file provides 100 channel creds options, the parser should not try to look up and check one by one. Only at the time this information is used, it would interpret it: going through the list to find the first one that gRPC supports. If none can be found, the gRPC client fails.

Reply:
It's confusing to see  "unknown type"  while type not provided. 

"unknown type" sounds more like user provides a type but it's not recognized. 

"with type unspecified" might be more clear.

Reply:
Sure, fixed the message.